awx/main/dispatch/worker/callback.py

```diff
@@ -152,6 +153,8 @@ def perform_work(self, body):
 
                 if body.get('event') == 'EOF':
                     try:
+                        if 'guid' in body:
+                            GuidMiddleware.set_guid(body['guid'])
```

Comment:
When we get the EOF event, set the guid from the message so that messages about job event processing include the guid